mkdocs/utils/__init__.py

```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
 
         branch = nested
         for part in parts:
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
+                dir = True
```

Comment:
Oh right, you can't modify `part` here because the later call to `find_or_create_node` needs the full filename. The general idea should still be doable though.